ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/ContractStates.kt

```diff
@@ -0,0 +1,86 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import java.security.PublicKey
+import java.util.*
+import kotlin.reflect.KClass
+
+/**
+ * Indicates the [Contract] that the current state belongs to.
+ */
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.CLASS)
+annotation class BelongsToContract(val value: KClass<out Contract>)
```

Comment:
This can't be used from Java, surely old one was Class?

Reply:
`KClass` works, `@InitiatedBy` used to be:

```kotlin
@Target(CLASS)
@MustBeDocumented
annotation class InitiatedBy(val value: KClass<out Flow<*>>)
```

Reply:
@mnesbit I tested it (although I need to re-add this to the current branch) - In Java, you just assign `Contract.class`, whereas in Kotlin you assign `Contract::class`. Must be some Kotlin magic that promotes a `Class<T>` to a `KClass<T>`.